crates/chain/src/tx_graph.rs

```diff
@@ -389,100 +385,62 @@ impl<A: Clone + Ord> TxGraph<A> {
                 0,
             ),
         );
-        self.determine_additions(&update)
-    }
-
-    /// Inserts the given [`TxOut`] at [`OutPoint`].
-    ///
-    /// This is equivalent to calling [`insert_txout_preview`] and [`apply_additions`] in sequence.
-    ///
-    /// [`insert_txout_preview`]: Self::insert_txout_preview
-    /// [`apply_additions`]: Self::apply_additions
-    pub fn insert_txout(&mut self, outpoint: OutPoint, txout: TxOut) -> Additions<A> {
-        let additions = self.insert_txout_preview(outpoint, txout);
-        self.apply_additions(additions.clone());
-        additions
+        let changeset = self.determine_changeset(&update);
```

Comment:
I see a recurrent pattern in these TxGraph method of calling `determine_changeset` and then `apply_changeset`. Can't we just replace these with `apply_update`?

Reply:
I think we could, but it's better if we do it in another PR, this one is pretty big already :)